lib/hashes.md

```diff
@@ -5,40 +5,46 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is a collection of objects, like an array, but they dont have to be numbers.  Hashes contain a set of 'keys' and 'values', where
+the values dont need to be numbers.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+The information used in a hash would be lists of items, to be set for a given value.  A ranking of different teams and their given value in different
+areas, such as strength, speed, debth, etc, or a list of ingredients on a pizza.
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Hashes and arrays are both lists, but hashes dont have to use numbers to index them, or describe them, they can be terms or definitions.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+A value from a hash can be retrieved by 'put'ing or 'print'ing the value like a variable.
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+A hash is built by defineing the hash, and puting given keys to be equal to or greater than the given value.
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+I am really struggling with this question, and the same question with refference to performing an opperation to every element in an array.
```

Comment:
Michael, I sent you an email with some more direction. But look into loops, and let me know if you have any specific questions. We can figure out the best path forwards from there.